crates/protocol/src/quests/mod.rs

```diff
@@ -118,6 +118,51 @@ impl QuestDefinition {
                         step.id.clone(),
                     ));
                 }
+
+                for action_item in &task.action_items {
+                    match &*action_item.r#type {
+                        "CUSTOM" => {
+                            if action_item.parameters.get("id").is_none() {
+                                return Err(QuestValidationError::ActionItemParametersNotValid(
+                                    "CUSTOM".to_string(),
+                                ));
+                            }
+                        }
+                        "LOCATION" => {
+                            if action_item.parameters.get("x").is_none()
+                                || action_item.parameters.get("y").is_none()
+                            {
+                                return Err(QuestValidationError::ActionItemParametersNotValid(
+                                    "LOCATION".to_string(),
+                                ));
+                            }
+                        }
+                        "EMOTE" => {
+                            if action_item.parameters.get("x").is_none()
+                                || action_item.parameters.get("y").is_none()
+                                || action_item.parameters.get("id").is_none()
+                            {
+                                return Err(QuestValidationError::ActionItemParametersNotValid(
+                                    "EMOTE".to_string(),
+                                ));
+                            }
+                        }
+                        "JUMP" => {
+                            if action_item.parameters.get("x").is_none()
+                                || action_item.parameters.get("y").is_none()
+                            {
+                                return Err(QuestValidationError::ActionItemParametersNotValid(
+                                    "JUMP".to_string(),
+                                ));
+                            }
+                        }
+                        _ => {
+                            return Err(QuestValidationError::ActionItemTypeNotValid(
+                                action_item.r#type.clone(),
+                            ));
+                        }
```

Comment:
I would leave it open to new actions, otherwise it would requiere a new release to accept new ones.

Reply:
I'd prefer to keep it as it is and have more control over this instead of leaving it free. In turn, the service and the documentation go hand in hand, since in the documentation we point out that these are the allowed actions beyond which we are able to accept more than this.